### ISSUE TITLE
FutureWarning in pm.summary(trace) when using Python 3.7

### ISSUE THREAD

**Issue by tuchandra** [pymc-devs/pymc3#3606]
Hello! This is my first time reporting an issue to PyMC3. This seems like a straightforward fix (Pandas gives a suggestion) but I am not sure if I have correctly identified the problem.

## Description of your problem
On Python 3.7, a `pd.concat` operation in `pymc3.stats.summary` causes Pandas to raise a FutureWarning.

**Please provide a minimal, self-contained, and reproducible example.**
```python
import pymc3 as pm

# the actual details of this model don't matter at all
with pm.Model() as model:
    p = pm.Uniform("p", lower=0, upper=1)
    obs = pm.Binomial("obs", n=10, p=p, observed=6)
    
    trace = pm.sample(100, tune=10)

pm.summary(trace)
```

**Please provide the full traceback.**
```python
/Users/chtu8001/miniconda3/envs/pymc/lib/python3.7/site-packages/pymc3/stats.py:982: 
FutureWarning: The join_axes-keyword is deprecated. Use .reindex or .reindex_like on 
the result to achieve the same functionality.
  axis=1, join_axes=[dforg.index])
```

**Please provide any additional information below.**

The offending line appears to be in [pymc3.stats](https://github.com/pymc-devs/pymc3/blob/master/pymc3/stats.py#L991) in the `summary` function. 

## Versions and main components

* PyMC3 Version: 3.6
* Theano Version: 1.0.4
* Python Version: 3.7
* Operating system: MacOS Mojave, 10.14.5
* How did you install PyMC3: conda

**Comment by ColCarroll** [comment]
I think you're right! Would you like to submit a fix to this? 

We do not test for warnings, so this would be just making the change suggested by pandas where you linked to it. I'm not great with pandas, so I'm not sure exactly *how* to implement the suggestion, but I'm happy to review!

**Comment by ColCarroll** [comment]
Fixed by #3608
